test/functional/Java8andUp/src/org/openj9/test/attachAPI/TestJps.java

```diff
@@ -0,0 +1,166 @@
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+@Test(groups = { "level.sanity" })
+public class TestJps extends AttachApiTest {
+
+	private static final String MISSING_ARGUMENT = "Missing argument: "; //$NON-NLS-1$
+	private static final String CHILD_IS_MISSING = "child is missing"; //$NON-NLS-1$
+	private static final String TEST_PROCESS_ID_MISSING = "Test process ID missing"; //$NON-NLS-1$
+	private static final String CHILD_PROCESS_DID_NOT_LAUNCH = "Child process did not launch"; //$NON-NLS-1$
+	private static final String JPS_COMMAND = "jps"; //$NON-NLS-1$
+	private static final String JPS_Class = "Jps"; //$NON-NLS-1$
+	boolean skipTest;
+	private String vmId;
+	
+	@Test(groups = { "level.sanity" })
+	public void testJpsSanity() throws IOException {
+		if (skipTest) {
+			return;
+		}
+		TargetManager tgtMgr = new TargetManager(TARGET_VM_CLASS, null);
+		assertTrue(CHILD_PROCESS_DID_NOT_LAUNCH, tgtMgr.syncWithTarget());
+		List<String> jpsOutput = runCommand();
+		assertTrue(TEST_PROCESS_ID_MISSING, search(vmId, jpsOutput).isPresent());
+		assertTrue("jps is missing", search(JPS_Class, jpsOutput).isPresent()); //$NON-NLS-1$
+		assertTrue(CHILD_IS_MISSING, search(tgtMgr.targetId, jpsOutput).isPresent());
+		tgtMgr.terminateTarget();
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testJpsPackageName() throws IOException {
+		final String WRONG_PKG_NAME = "Wrong package name: "; //$NON-NLS-1$
+		if (skipTest) {
+			return;
+		}
+		TargetManager tgtMgr = new TargetManager(TARGET_VM_CLASS, null);
+		assertTrue(CHILD_PROCESS_DID_NOT_LAUNCH, tgtMgr.syncWithTarget());
+		List<String> jpsOutput = runCommand(Collections.singletonList("-l")); //$NON-NLS-1$
+		Optional<String> searchResult = search(tgtMgr.targetId, jpsOutput);
+		assertTrue(CHILD_IS_MISSING, search(tgtMgr.targetId, jpsOutput).isPresent());
+		assertTrue(WRONG_PKG_NAME + searchResult.get(), searchResult.get().contains(TargetVM.class.getPackage().getName()));
+		tgtMgr.terminateTarget();
+	}
+
+	@Test(groups = { "level.extended" })
+	public void testJpsIdOnly() throws IOException {
+		final String SPURIOUS_OUTPUT = "Spurious output: "; //$NON-NLS-1$
+		if (skipTest) {
+			return;
+		}
+		TargetManager tgtMgr1 = new TargetManager(TARGET_VM_CLASS, null);
+		TargetManager tgtMgr2 = new TargetManager(TARGET_VM_CLASS, "SomeRandomId"); //$NON-NLS-1$
+		assertTrue(CHILD_PROCESS_DID_NOT_LAUNCH, tgtMgr1.syncWithTarget());
+		List<String> jpsOutput = runCommand(Collections.singletonList("-q")); //$NON-NLS-1$
+		Optional<String> searchResult = search(vmId, jpsOutput);
+		assertTrue(TEST_PROCESS_ID_MISSING, searchResult.isPresent());
+		assertTrue(SPURIOUS_OUTPUT + searchResult.get(), searchResult.get().equals(vmId));
+		searchResult = search(tgtMgr1.targetId, jpsOutput);
+		assertTrue(CHILD_IS_MISSING, search(tgtMgr1.targetId, jpsOutput).isPresent());
+		assertTrue(SPURIOUS_OUTPUT + searchResult.get(), searchResult.get().equals(tgtMgr1.targetId));
```

Comment:
Use of `assertEquals` is preferred because when it fails it present both values for comparison.